mypyc/lib-rt/setup.py

```diff
@@ -25,9 +25,56 @@
     "pythonsupport.c",
 ]
 
+EXTRA_FLAGS_PER_COMPILER_TYPE_PER_PATH_COMPONENT = {
+    "unix": {
+        "base64/arch/ssse3": ["-mssse3", "-DBASE64_WITH_SSSE3"],
+        "base64/arch/sse41": ["-msse4.1", "-DBASE64_WITH_SSE41"],
+        "base64/arch/sse42": ["-msse4.2", "-DBASE64_WITH_SSE42"],
+        "base64/arch/avx2": ["-mavx2", "-DBASE64_WITH_AVX2"],
+        "base64/arch/avx": ["-mavx", "-DBASE64_WITH_AVX"],
+    },
+    "msvc": {
+        "base64/arch/sse42": ["/arch:SSE4.2", "/DBASE64_WITH_SSE42"],
+        "base64/arch/avx2": ["/arch:AVX2", "/DBASE64_WITH_AVX2"],
+        "base64/arch/avx": ["/arch:AVX", "/DBASE64_WITH_AVX"],
+    },
+}
+
+ccompiler.CCompiler.__spawn = ccompiler.CCompiler.spawn  # type: ignore[attr-defined]
 X86_64 = platform.machine() in ("x86_64", "AMD64", "amd64")
 
 
+def spawn(self, cmd, **kwargs) -> None:  # type: ignore[no-untyped-def]
```

Comment:
This looks similar if not the same, any particular reason why not have it in a shared location, or is this because of the fact these are `setup` files?